Package 5.4.0 as akmod

ffgiff opened the pull request:
Add specfiles to package evdi as an akmod, with separate packages for
DisplayLink Manager and the Fedora-specific config files.

Tie them all together with a metapackage to make installation simpler.

Since upstream evdi wasn't tagged properly we need to point to a
commit id instead of the tag.

Add install test on Travis showing that Centos build fails (#13).
Fixes #81, replaces previous PR #97 as rework had confused GitHub.

ffgiff committed:
Package 5.4.0 as akmod

Add specfiles to package evdi as an akmod, with separate packages for
DisplayLink Manager and the Fedora-specific config files.

Tie them all together with a metapackage to make installation simpler.

rapgro commented:
travis-ci shows an issue with docker authentication, wrong password?

ffgiff commented:
> travis-ci shows an issue with docker authentication, wrong password?

That's #164 still showing up -- for security, the environment variables are blocked for PRs coming from outside the project.

I suppose you could fix this by only logging into docker if username is set in the env.

The Travis job succeeds if it can find a valid username/password, e.g. in the forked repo: https://github.com/ffgiff/displaylink-rpm/runs/2442562670

elguero commented:
I have attempted to fix this on pull requests but failed.  Yesterday, I realized that I had inverted the logic.  If someone would like to double-check me, it is here: https://github.com/displaylink-rpm/displaylink-rpm/commit/52617e197ce90c18302bbbb795643903d9ec5c95

@ffgiff Could you try merging that commit into your branch and see if that resolves the issue of PR requests not building?  Thanks

ffgiff commented:
Re-based, updated to use re-tagged upstream evdi, and fixed rpmlint warnings against the spec files. There are still warnings on the rpms, but I don't think they're fixable.

ssaavedra commented on akmod-specs/displaylink.spec:
```diff
@@ -0,0 +1,138 @@
+%{_presetdir}/95-displaylink.preset
+
+%post config
+%systemd_post displaylink-service
```
Should this be a dot instead? `displaylink.service` instead of `displaylink-service`?

ssaavedra commented on akmod-specs/displaylink.spec:
```diff
@@ -0,0 +1,138 @@
+%dir /var/log/%{name}/
+%{_prefix}/lib/systemd/system-sleep/displaylink.sh
+
+%package config
```
Also what's the rationale to separate the "config" package from the "manager" package?
It seems that one cannot live without the other anyway.

ffgiff replied:
Config is the files originating from and licensed by this Github project, Manager is the files originating from and licensed by Synaptics/DisplayLink. Since we need multiple packages anyway for the akmod build, splitting config from manager means you get one license per package instead of multiple licenses covering different files in the same package.

ssaavedra replied:
Hm.., interesting. This could mean that if RPMfusion ever picks this, they could more easily put parts into their separate free/non-free repos; although at the moment it would be useless for someone on the free repo to want to install only the free packages (even if DisplayLink Inc argues evdi can be useful on its own, it hasn't happened so far).

ssaavedra replied:
Fair enough :-) 👍

ssaavedra commented on akmod-specs/displaylink.spec:
```diff
@@ -0,0 +1,138 @@
+cp -a 99-displaylink.rules %{buildroot}%{_udevrulesdir}
+cp -a 20-displaylink.conf %{buildroot}%{_datadir}/X11/xorg.conf.d
+
+%changelog
```
Shouldn't we keep the changelog from former versions, and maybe add an `Obsoletes:` for the old package?

ffgiff replied:
I'm not sure on that. It's largely based on the DKMS displaylink.spec (so yes), but it's also a separate file (so no). Overall, I think it depends on whether this replaces the DKMS displaylink package or not. I suppose you could 'upgrade' or 'downgrade' between the DKMS and akmod, but I haven't and cannot test that (I'm running Silverblue, so DKMS isn't possible, hence this PR).

ssaavedra commented:
Thanks for the effort! Would you please take a look at my comments and see if it makes sense to you? I think splitting evdi and displaylink this way would be interesting.

However, I'm a bit concerned that given that Github Releases is not an actual rpm repository, downloading more files is more cumbersome than the current approach. Maybe we should tie this to a copr repository as a second step if we split the packaging.

ffgiff committed:
Support 5.13 and 5.14

ffgiff committed:
Support 5.15

ffgiff committed:
Update for kernel 5.16

ffgiff committed:
Catch-up to master

AltF02 commented:
Any update on this?

ffgiff committed:
Update akmod specs for 5.5

ffgiff committed:
Merge remote-tracking branch 'upstream/master' into akmod-5.6-beta

ffgiff committed:
Update akmod specs for 5.6

ffgiff committed:
Merge remote-tracking branch 'upstream/master' into HEAD

ffgiff committed:
Use RHEL macros instead of CentOS, add licence identifiers

ffgiff committed:
Catch up to upstream master

ffgiff committed:
Update to 5.6.1/1.12.0

ffgiff committed:
Escape urls properly

leleobhz commented:
Hello!

Just a cent in this thread: This PR is required for Fedora Silverblue [1], so for Silverblue users, evdi requires manual build at runtime.

[1]: https://github.com/coreos/rpm-ostree/issues/1091#issuecomment-438201426

leleobhz commented:
A workarround is use DKMS ans since instalation of module will fail, override `displaylink-driver.service` to allow manual module loading

```
# /etc/systemd/system/displaylink-driver.service.d/override.conf
[Unit]
After=dkms.service

[Service]
ExecStartPre=
ExecStartPre=-/usr/bin/sh -c 'find /var/lib/dkms/evdi/*/%v -name *ko* -exec insmod {} \\;'
ExecStopPost=
ExecStopPost=-/sbin/rmmod -f evdi
```

ffgiff commented:
> A workarround is use DKMS ans since instalation of module will fail, override `displaylink-driver.service` to allow manual module loading
> 
> ```
> # /etc/systemd/system/displaylink-driver.service.d/override.conf
> [Unit]
> After=dkms.service
> 
> [Service]
> ExecStartPre=
> ExecStartPre=-/usr/bin/sh -c 'find /var/lib/dkms/evdi/*/%v -name *ko* -exec insmod {} \\;'
> ExecStopPost=
> ExecStopPost=-/sbin/rmmod -f evdi
> ```

The whole point of this PR is to use akmod instead of DKMS, primarily so that DisplayLink devices can be used on Silverblue. This has been working for me for around four years now, see the original PR #97 . No special workarounds are needed, just build the rpms using the provided ci scripts and install.

leleobhz commented:
> > A workarround is use DKMS ans since instalation of module will fail, override `displaylink-driver.service` to allow manual module loading
> > ```
> > # /etc/systemd/system/displaylink-driver.service.d/override.conf
> > [Unit]
> > After=dkms.service
> > 
> > [Service]
> > ExecStartPre=
> > ExecStartPre=-/usr/bin/sh -c 'find /var/lib/dkms/evdi/*/%v -name *ko* -exec insmod {} \\;'
> > ExecStopPost=
> > ExecStopPost=-/sbin/rmmod -f evdi
> > ```
> 
> The whole point of this PR is to use akmod instead of DKMS, primarily so that DisplayLink devices can be used on Silverblue. This has been working for me for around four years now, see the original PR #97 . No special workarounds are needed, just build the rpms using the provided ci scripts and install.

There is no need to use workarounds if using akmods, but its needed if using main from repo since this PR wasn't merged yet. This is a temporary way to get my monitor working while this PR doesn't get merged and I prefer this repo uses akmods instead dkms.

I need to clarify my comment with the following info: This workaround is meant to be used only before this PR got merged AND if someone wants to use main repo while this PR still open.

Thank you for you point!

leleobhz commented:
> Thanks for the effort! Would you please take a look at my comments and see if it makes sense to you? I think splitting evdi and displaylink this way would be interesting.
> 
> However, I'm a bit concerned that given that Github Releases is not an actual rpm repository, downloading more files is more cumbersome than the current approach. Maybe we should tie this to some kind of rpm repository (although we cannot use copr since DisplayLinkManager is not Free Software) as a second step if we split the packaging.

@ssaavedra The question about split RPM repositories is the question holding this PR?

BlauerHunger commented:
Any chance this could be merged? If not, why? 
I'd like to have a clean install of Silverblue on my notebook without having to rely on ugly workarounds like the one mentioned in this thread but instead have displaylink+evdi installed in a proper way.

bi0shacker001 commented:
Same here. I rely on displaylink pretty heavily for work, so this issue means I have to dual-boot linuxes to use my dock (as that workaround hasn't been working reliably for me)

dv-parkingspotz commented:
Any update on this?

ullebe1 commented:
For those interested I've just updated @ffgiff's work to be based on DisplayLink 6.1.0 and libevdi 1.14.8.
It works for me, but as I'm not very familiar with RPM packaging I can't say if I've made any mistakes.

This can be found here: https://github.com/ullebe1/displaylink-rpm/tree/new-akmod

ullebe1 commented:
> @ullebe1 I've tried your branch but it makes no difference for me (installing on QubesOS, sys-usb). I built the .src.rpm using `make`, was that correct or do I need a different command for your version?

I built it using the commands in this [file](https://github.com/ullebe1/displaylink-rpm/blob/new-akmod/akmod-specs/ci/fedora.sh), adjusted to run in a Distrobox container. Not sure which set of instructions are appropriate for QubesOS as I'm not familiar with its lineage, I'd probably try those or the ones for CentOS in the same directory.

If @ffgiff isn't interested in being the one to carry this over the finish line I might try to hook this up to the proper GitHub Actions to build release artifacts just like the normal variants and then do my own PR with the updated version. That should at least make it easier to get something useful before it is merged, as one could just clone my fork and run the action in something like Act (if I don't just let them run in my forked repo).

ffgiff commented:
> > @ullebe1 I've tried your branch but it makes no difference for me (installing on QubesOS, sys-usb). I built the .src.rpm using `make`, was that correct or do I need a different command for your version?
> 
> I built it using the commands in this [file](https://github.com/ullebe1/displaylink-rpm/blob/new-akmod/akmod-specs/ci/fedora.sh), adjusted to run in a Distrobox container. Not sure which set of instructions are appropriate for QubesOS as I'm not familiar with its lineage, I'd probably try those or the ones for CentOS in the same directory.
> 
> If @ffgiff isn't interested in being the one to carry this over the finish line I might try to hook this up to the proper GitHub Actions to build release artifacts just like the normal variants and then do my own PR with the updated version. That should at least make it easier to get something useful before it is merged, as one could just clone my fork and run the action in something like Act (if I don't just let them run in my forked repo).

@ullebe1 please feel free to take this forward, I replaced my DisplayLink adapter after it failed and so I no longer have a need for this, or the hardware to test and maintain these packages.

ullebe1 commented:
> @ullebe1 please feel free to take this forward, I replaced my DisplayLink adapter after it failed and so I no longer have a need for this, or the hardware to test and maintain these packages.

Thank you for clarifying and for your work so far, I'll attempt to pick it up and see if we can get it the last bit of the way.

regulad commented:
@ullebe1 I updated your branch to Driver 6.1.1, allowing it to support Kernel 6.14-based distros like Fedora 42.

If there is anything more I can do to help this get put on RPMFusion, please feel free to reach out here or at my git email.

ullebe1 commented:
> @ullebe1 I updated your branch to Driver 6.1.1, allowing it to support Kernel 6.14-based distros like Fedora 42.
> 
> If there is anything more I can do to help this get put on RPMFusion, please feel free to reach out here or at my git email.

Thank you!

This is a bit embarrassing: I made the same change when I upgraded to Fedora 42, but totally forgot to commit and push it when it was working as I was just happy that it did.

I'll try comparing our changes to see if we did it the same way and if I can learn something from your approach.

MusterGit commented:
Wouldn't this be a nice birthday present for Fedora 44? :smile:
